codec/src/main/java/io/netty/handler/codec/LineBasedFrameDecoder.java

```diff
@@ -25,6 +25,10 @@
  * A decoder that splits the received {@link ByteBuf}s on line endings.
  * <p>
  * Both {@code "\n"} and {@code "\r\n"} are handled.
+ * <p>
+ * The byte stream is expected to be in UTF-8 character encoding or ASCII, as current implementation
+ * uses direct {@code byte} to {@code char} cast.
+ * <p>
```

Comment:
`byte` to `char` cast sounds a bit misleading in the context of UTF-8, which is a [variable-length encoding](https://tools.ietf.org/html/rfc3629#section-4). Though on a quick glance it looks like all of the decoders only operate on characters which do have the same one-byte representation in ASCII and UTF-8.

Reply:
That's right. All decoders are using low range (<=0x7F) codepoints (e.g., "<>/="\r\n{}[]:") in their implementations, and in UTF-8 the same bytes are not appear in second, third or fourth byte of a multibyte codepoint representation. See [URT-8](https://en.wikipedia.org/wiki/UTF-8) description. So all decoders are going to handle UTF-8 and ASCII properly. 

Perhaps the comment could be further clarified by going "as current implementation uses direct `code` to `byte` cast and then compares it with low range ASCII characters like `<` `>`, `\r` `\n`, etc...

Or we can remove the reference to cast at all... Though I see some value in having this particular implementation disclosure.

Up to you.

Reply:
I think your clarification makes sense. :+1: It now also accounts for other ASCII compatible character sets than just UTF-8 (e.g. [ISO 8859](https://en.wikipedia.org/wiki/ISO_8859)).